src/data_source/storage/sql/queries/explorer.rs

```diff
@@ -105,6 +105,164 @@ where
     }
 }
 
+lazy_static::lazy_static! {
+    static ref GET_BLOCK_SUMMARIES_QUERY_FOR_LATEST: String = {
```

Comment:
Are these static variables defined just for readability? I see these queries being used only once.

Reply:
and do we have tests for all these queries?

Reply:
They are primarily being defined as a static reference for a couple of reasons and advantages.

- To ensure that they are indeed a static, unchanging string.

Query strings **should** be largely unchanging.  Otherwise you open yourself up to the potential for SQL injection attacks.  Consider what was being done before:

```rust
  BlockIdentifier::Hash(hash) => format!(
    "SELECT {BLOCK_COLUMNS}
        FROM header AS h
        JOIN payload AS p ON h.height = p.height
        WHERE h.hash = {}
        ORDER BY h.height DESC 
        LIMIT 1",
    query.bind(hash.to_string())?,
  )
```

 At first glance this seems to be roughly the same thing.  But, taking a closer look, the entire query string is in a `format` macro with a variable replacement / injection being provided by the result of the `query.bind`.  I don't believe this is the intended or correct behavior.  Additionally, attempts to split this up resulted in a difficult to resolve lifetime issue.

- This allows the query strings to be defined once, and then referenced multiple times.

This cleans up a lot of lifetime issues, as the strings are static, and outlive their async blocks.

- It prevents the string from having to be re-computed / formatted for every invocation of the query, saving memory.

- The static unchanging string also allows for `sqlx`'s query caching to be guaranteed to take advantage of its internal prepared statements for re-use.

Reply:
During the refactor of this commit: 8f64921843d4cb27ad0e88b86a789869c369975a, the separation of the query string definition, and the query parameters were ultimately modified, leading to the potential for an SQL injection, as we are directly inserting a string coming from the end-user.

Reply:
These queries do get tested indirectly at this symbol:
`explorer::test::validate`.

Reply:
The bind() method of the QueryBuilder returns `self.arguments.len()`  so basically, it provides a placeholder for the query and stores the arguments in its struct.

Reply:
Oh I see.  I made an assumption as to what this was returning because I couldn't think of any other reason why the result of `bind` would return a `Result<String>`.  So this currently returns a placeholder representation to put into query strings.

I do think that actually utilizing the `bind` call in this way with the `format!` macro is best avoided.

The result of the `bind` function could potentially be modified in the future without changing the API to return a different `String` value in the future which would not cause a breaking change in terms of the API.  I do think that this is unlikely, but the possibility for it to happen is non-zero.

Additionally, since there is a `format!` macro, if in the future another parameter is added to these queries for whatever reason, we leave the possibility of a developer just adding it to the query string via a `format` parameter instead of through the `QueryBuilder` itself.  While we could argue that this would surely be caught in review, I think it's probably best to avoid that possibility from the get go.

Reply:
I agree that this bind() is confusing and should be changed.  I am thinking we store the string with placeholders inside the QueryBuilder and bind() can return the QueryBuilder or it might just take &mut self

Reply:
https://github.com/EspressoSystems/hotshot-query-service/issues/763